[feature][KLTB002-14736] Add Jira support

drazenzen opened the pull request:
## Description
This Pull Request add Jira support to `kolibree_output`.

## Preflight Checklist

- [x] No warnings or linting issues have been introduced

##
#### Jira ticket
For more details check the [JIRA ticket](https://kolibree.atlassian.net/browse/KLTB002-14736).

#### How it works
Jira issue (summary field) is retrieved and added to changelog ouput in case Github description is missing.

drazenzen committed:
Add Jira support

georgescumihai commented on src/gitchangelog/gitchangelog.py:
```diff
@@ -1394,6 +1421,29 @@ def kolibree_output(data, opts={}):
+                basic_auth=(jira_username, jira_apitoken)
+            )
+        except Exception:
+            die("Unable to connect to Jira")
```
This is already handled in `get_jira`

drazenzen replied:
Yes, that is a leftover, cleaned up.

georgescumihai commented:
We should use Jira as the first choice for the changelog entry title and github as a backup in case the git commit doesn't have a jira ticket.
If jira is used, add the jira ticket url to the changelog entry for `kolibree` output engine. using the following format
`[JIRA-Ticket](jira_ticket_url) Jira ticket summary`

drazenzen committed:
Jira as source of truth

drazenzen committed:
Cleanup

drazenzen committed:
Add Jira url in entry title

drazenzen committed:
Add square brackets around Jira link

georgescumihai commented on src/gitchangelog/gitchangelog.py:
```diff
@@ -1418,10 +1460,24 @@ def render_version(version):
+        if ticket:
+            try:
+                issue = jira.issue(ticket, fields="summary")
+                subject = "[[{}]]({}) {}".format(
```
there is one pair of extra brackets 
Output:
`- [[KLTB002-14238]](https://kolibree.atlassian.net/browse/KLTB002-14238) Setup repository for changelog`

Expected:
`- [KLTB002-14238](https://kolibree.atlassian.net/browse/KLTB002-14238) Setup repository for changelog`

```suggestion
                subject = "[{}]({}) {}".format(
```

drazenzen replied:
Ok, thought that we want Jira link visually inside square brackets :) Will remove them.

drazenzen replied:
Updated. Lets please keep notes of any other formatting glitch/issue/... and tackle it in following formatting task.

georgescumihai reviewed (changes requested):
Nice work, I found some small issues but should be fast to fix.

There is a small issue with the output for jira, with the extra brackets
Add `requirements.txt` for the Jira package and maybe for Github as well. if you do not install them the script will fail.

drazenzen committed:
Add requirements.txt and update setup.py

drazenzen committed:
Remove surplus brackets

drazenzen committed:
Protect searching for Jira ticket if regex is not set

drazenzen commented:
> Nice work, I found some small issues but should be fast to fix.
> 
> There is a small issue with the output for jira, with the extra brackets
> Add `requirements.txt` for the Jira package and maybe for Github as well. if you do not install them the script will fail.

Added root `requirements.txt` with required libs (`PyGithub` and `jira`) and updated `setup.py` if we are going to install/use package through `setup.py`

lookashc commented:
Made tests on android monorepo, config available on branch https://github.com/kolibree-git/android-monorepo/tree/changelog
Command: `gitchangelog sdk/7.1.0..sdk/9.0.0`
Results: [sdk_9_0_0.txt](https://github.com/kolibree-git/gitchangelog/files/6042169/sdk_9_0_0.txt)
Github diff for comparison: https://github.com/kolibree-git/android-monorepo/compare/sdk/7.1.0...sdk/9.0.0

Jira integration works 🎉 

I noticed we are making the choice between sections based on `section_regexps` in config - while we should do that based on Jira ticket type, as defined here https://kolibree.atlassian.net/wiki/spaces/SOF/pages/950468787/Generator+Requirements. Currently, all Jira-related items are in `Features` section while they should be spread between different sections, based on the associated ticket type. But I guess this is for the next step - @georgescumihai can you confirm?

✅  Great job @drazenzen 🥇

georgescumihai commented:
Yes, there are future stories to group tickets based on Jira issue type and exclude the tickets which are not done. 
The scope of this story is only for the Jira integration.
